[2024-05-08T10:02:45.537+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-05-08T09:40:00+00:00 map_index=6 [queued]>
[2024-05-08T10:02:45.583+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-05-08T09:40:00+00:00 map_index=6 [queued]>
[2024-05-08T10:02:45.584+0000] {taskinstance.py:2193} INFO - Starting attempt 2 of 4
[2024-05-08T10:02:45.634+0000] {taskinstance.py:2217} INFO - Executing <Mapped(_PythonDecoratedOperator): ETL_job> on 2024-05-08 09:40:00+00:00
[2024-05-08T10:02:45.679+0000] {standard_task_runner.py:60} INFO - Started process 858252 to run task
[2024-05-08T10:02:45.694+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'Units_statistics_DAG', 'ETL_job', 'scheduled__2024-05-08T09:40:00+00:00', '--job-id', '41371', '--raw', '--subdir', 'DAGS_FOLDER/Units_dag.py', '--cfg-path', '/tmp/tmpxxblkzwu', '--map-index', '6']
[2024-05-08T10:02:45.720+0000] {standard_task_runner.py:88} INFO - Job 41371: Subtask ETL_job
[2024-05-08T10:02:45.957+0000] {task_command.py:423} INFO - Running <TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-05-08T09:40:00+00:00 map_index=6 [running]> on host a09da322e808
[2024-05-08T10:02:46.190+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='abeer-araby' AIRFLOW_CTX_DAG_ID='Units_statistics_DAG' AIRFLOW_CTX_TASK_ID='ETL_job' AIRFLOW_CTX_EXECUTION_DATE='2024-05-08T09:40:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-05-08T09:40:00+00:00'
[2024-05-08T10:02:46.191+0000] {Units_dag.py:98} INFO - i: 60000, chunk: 10000
[2024-05-08T10:02:46.192+0000] {Units_dag.py:99} INFO - child token is::: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2FwaS5hZmFxeS5zYS9hdXRoL2xvZ2luX2FzIiwiaWF0IjoxNzE1MTYyNDA0LCJleHAiOjE3MTc3NTQ0MDQsIm5iZiI6MTcxNTE2MjQwNCwianRpIjoiMGR6ZVRzNjR1Zk1VT1dCcyIsInN1YiI6IjVkNjNlNTFlMDNlZDUyOTJhNDJjMjRlNiIsInBydiI6IjI3MGVjZmM3ZWEzZWQ5MzdlYTg0OTM2MmEzYTUwOTEwYzZkOGNlNGYiLCJsb2dpbl9hc19pZCI6IjY1N2YwYWQ3M2Y5NzcyOTVjZjA1NTc2YSJ9.zAlCNedYcuchsVzdj1fvtJF31uHFK7r6EE--HZgB-p4
[2024-05-08T10:03:05.719+0000] {Units_dag.py:103} INFO - {'_id': '63302127973f5950ea3cceb7', 'icon': 'icons/icon.png', 'driver_id': '', 'tailer_id': '', 'profile': {'vehicle_type': 'ACCENT', 'measurement_type': 'count', 'max_capacity': 0, 'vin': '', 'plate_number': '', 'start_operation': '', 'contract_milage': '', 'opening_milage': '', 'model': '', 'fuel_type': '91', 'year': 0, 'seats': 2}, 'device_password': '***', 'device_type': 'gps', 'gateway': False, 'name': '5453UJR', 'operation_code': '', 'device_gateway': '', 'business_types': {'waste': False, 'transportation': True, 'water': False}, 'imei': '860665057470014', 'sim_number': '+966830020211446', 'sim_serial': '996641521115987406', 'device_serial': '860665057470014', 'device': 'bceIotm', 'user_id': '5e68ea7a03ed528a611e0f38', 'created_by': '5e6655bf03ed52e1ab79d301', 'updated_at': '2024-04-30 10:01:12', 'created_at': '2022-09-25 12:36:39', 'shipment_info': {'product_category': None, 'truck_type': None, 'shipment_type': None, 'shipment_payment_type': None, 'transporter_type': None, 'operating_area': None, 'weight_kilograms_per_day': None, 'kms_per_day': None, 'shipment_trip_drive_hrs_day': None, 'allowable_delay_drive_hrs_day': None, 'expected_shipment_trip_day': None, 'car_pay_per_day': None, 'exceeding_temperature_limit_per_day': None, 'temperature_upper_limit': None, 'exceeding_humidity_limit_per_day': None, 'humidity_upper_limit': None}, 'skip_sim_information': '', 'custom_fields': [], 'cctv': {'dvr_id': '', 'channels': 1}, 'creator': 'amr.ahmed', 'company': '', 'owner': 'shift_co', 'last_update': {'dtt': 1715162458000, 'dts': 1715162463034, 'spd': 0, 'ang': 0, 'alt': 634, 'chPrams': {'acc': {'v': 0, 'cdt': 1715119115000}, 'sensor_8192': {'v': 0, 'cdt': 1715116819000}, 'sensor_16386': {'v': 0, 'cdt': 1712942675000}, 'sensor_16385': {'v': 11608, 'cdt': 1715119077000}, 'fw_name': {'v': 'FF0_01_122', 'cdt': 1712942675000}, 'ePwrV': {'v': 12224, 'cdt': 1715162458000}, 'sensor_8199': {'v': 33, 'cdt': 1715116819000}, 'sensor_16387': {'v': 5412400, 'cdt': 1715116758000}, 'sensor_153': {'v': 0, 'cdt': 1697375936000}, 'sensor_12300': {'v': 0, 'cdt': 1715119115000}, 'war_id': {'v': 145, 'cdt': 1715162237000}, 'protocol': {'v': 'MQTT', 'cdt': 1714827578000}, 'sensor_97': {'v': 0, 'cdt': 1712942675000}, 'sensor_98': {'v': 0, 'cdt': 1715119071000}, 'sensor_99': {'v': 0, 'cdt': 1715119154000}, 'sensor_12292': {'v': 4071, 'cdt': 1715162458000}, 'spd': {'v': 0, 'cdt': 1715072514000}, 'sat': {'v': 12, 'cdt': 1715116279000}, 'sensor_103': {'v': 0, 'cdt': 1712942675000}, 'fun_id': {'v': 0, 'cdt': 1715162237000}, 'sensor_102': {'v': 0, 'cdt': 1712942675000}, 'sensor_8200': {'v': 134, 'cdt': 1715119115000}, 'sensor_101': {'v': 0, 'cdt': 1715114345000}, 'sensor_100': {'v': 0, 'cdt': 1715118969000}, 'hdop': {'v': 5, 'cdt': 1715160652000}, 'sensor_12299': {'v': 0, 'cdt': 1715116788000}, 'sensor_49152': {'v': 'GPRS S1=1 (block) command executed', 'cdt': 1715119170000}, 'sensor_40962': {'v': 0, 'cdt': 1715116785000}, 'sensor_49156': {'v': 'SMS accent lock/unlo', 'cdt': 1712942675000}, 'sensor_40961': {'v': 0, 'cdt': 1715116788000}}, 'prms': {'sensor_40962': 0, 'sensor_40961': 0, 'sensor_16386': 0, 'sensor_16387': 5412400, 'sensor_16385': 11608, 'sensor_12300': 0, 'ePwrV': 12224, 'sensor_12299': 0, 'sensor_12292': 4071, 'sensor_153': 0, 'sensor_8192': 0, 'sensor_99': 0, 'sensor_97': 0, 'sensor_8200': 134, 'sensor_103': 0, 'sensor_98': 0, 'sensor_102': 0, 'sensor_101': 0, 'acc': 0, 'spd': 0, 'hdop': 5, 'sat': 12, 'sensor_8199': 33, 'sensor_100': 0, 'protocol': 'MQTT'}, 'devPrms': ['sensor_40962', 'sensor_40961', 'sensor_16386', 'sensor_16387', 'sensor_16385', 'sensor_12300', 'ePwrV', 'sensor_12299', 'sensor_12292', 'sensor_153', 'sensor_8192', 'sensor_99', 'sensor_97', 'sensor_8200', 'sensor_103', 'sensor_98', 'sensor_102', 'sensor_101', 'acc', 'spd', 'hdop', 'sat', 'sensor_8199', 'sensor_100', 'protocol', 'fun_id', 'war_id', 'fw_name'], 'old_params': {'fun_id': 0, 'war_id': 145, 'fw_name': 'FF0_01_122', 'protocol': 'MQTT', 'sensor_40962': 0, 'sensor_40961': 0, 'sensor_16386': 0, 'sensor_16387': 5412400, 'sensor_16385': 11608, 'sensor_12300': 0, 'ePwrV': 12188, 'sensor_12299': 0, 'sensor_12292': 4078, 'sensor_153': 0, 'sensor_8192': 0, 'sensor_99': 0, 'sensor_97': 0, 'sensor_8200': 134, 'sensor_103': 0, 'sensor_98': 0, 'sensor_102': 0, 'sensor_101': 0, 'acc': 0, 'spd': 0, 'hdop': 5, 'sat': 12, 'sensor_8199': 33, 'sensor_100': 0}, 'vLoc': 1, 'vCon': 1, 'ip': '109.83.250.207', 'rl': 0, 'trl': 27062, 'mdu': 0, 'tmdu': 1085026, 'eidl': 0, 'teidl': 1990867, 'fc': 0, 'tfc': 1.68, 'acc': 0, 'accOn': 0, 'accOff': 43320, 'sdu': 221, 'tsdu': 43343, 'lat': 24.69975471496582, 'lng': 46.661746978759766}, 'counters': {'odometer_type': 'canbus', 'odometer': 27062, 'engine_hours_type': 'canbus', 'engine_hours': 0, 'last_acc': 0}, 'is_wasl_connected': 'Not Connected', 'iconUrl': 'https://api.afaqy.sa/images/units/icons/icon.png', 'hasCCTV': False}
[2024-05-08T10:03:06.315+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk1.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-05-08T10:03:06.377+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk2.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-05-08T10:03:06.419+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk3.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-05-08T10:03:06.447+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dbelk1.afaqy.local'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2024-05-08T10:03:06.523+0000] {_transport.py:335} INFO - GET https://dbelk1.afaqy.local:9200/_cluster/health [status:200 duration:0.102s]
[2024-05-08T10:03:06.933+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dbelk2.afaqy.local'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2024-05-08T10:03:06.949+0000] {_transport.py:335} INFO - PUT https://dbelk2.afaqy.local:9200/units_statistics_***/_bulk [status:429 duration:0.035s]
[2024-05-08T10:03:06.950+0000] {_node_pool.py:249} WARNING - Node <Urllib3HttpNode(https://dbelk2.afaqy.local:9200)> has failed for 1 times in a row, putting on 1 second timeout
[2024-05-08T10:03:06.950+0000] {_transport.py:427} WARNING - Retrying request after non-successful status 429 (attempt 0 of 3)
[2024-05-08T10:03:06.951+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dbelk1.afaqy.local'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2024-05-08T10:03:07.121+0000] {_transport.py:335} INFO - PUT https://dbelk1.afaqy.local:9200/units_statistics_***/_bulk [status:200 duration:0.170s]
[2024-05-08T10:03:07.397+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/Units_dag.py", line 111, in ETL_job
    Elk_client.elk_pushData(Globals.index_name,units_list_update)
  File "/opt/airflow/dags/units_statistics/elk_manager.py", line 69, in elk_pushData
    helpers.bulk(self.elk_client, docs_list, index=index_name, chunk_size=batch)
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 521, in bulk
    for ok, item in streaming_bulk(
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 436, in streaming_bulk
    for data, (ok, info) in zip(
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 355, in _process_bulk_chunk
    yield from gen
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 274, in _process_bulk_chunk_success
    raise BulkIndexError(f"{len(errors)} document(s) failed to index.", errors)
elasticsearch.helpers.BulkIndexError: 10000 document(s) failed to index.
[2024-05-08T10:03:07.413+0000] {taskinstance.py:1149} INFO - Marking task as UP_FOR_RETRY. dag_id=Units_statistics_DAG, task_id=ETL_job, map_index=6, execution_date=20240508T094000, start_date=20240508T100245, end_date=20240508T100307
[2024-05-08T10:03:07.433+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 41371 for task ETL_job (10000 document(s) failed to index.; 858252)
[2024-05-08T10:03:07.474+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-05-08T10:03:07.503+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
